File: java-soa/inventory-service/src/main/java/com/ecommerce/inventory/infrastructure/web/dto/InventoryResponse.java
package com.ecommerce.inventory.infrastructure.web.dto;

import java.time.LocalDateTime;
import java.util.UUID;

public record InventoryResponse(
        UUID productId,
        String productName,
        int quantityAvailable,
        int quantityReserved,
        LocalDateTime updatedAt
) {}

File: dotnet-soa/InventoryService/Infrastructure/Web/Dto/InventoryResponse.cs
namespace InventoryService.Infrastructure.Web.Dto;

/// <summary>
/// DTO for inventory API responses.
/// </summary>
public record InventoryResponse
{
    public Guid Id { get; init; }
    public Guid ProductId { get; init; }
    public string ProductName { get; init; } = string.Empty;
    public int Quantity { get; init; }
    public int ReservedQuantity { get; init; }
    public int AvailableQuantity { get; init; }
    public DateTime CreatedAt { get; init; }
    public DateTime UpdatedAt { get; init; }
}
